### PR TITLE
Add Documnet Icon & Attachment components

### PR DESCRIPTION
**Pull request by nicklera1**
**Description of changes:**
Added Document Icon & MessageAttachement components.

*** Preview ***

Screen Shot 2020-10-22 at 9 51 25 AM

**Testing**
1. Have you successfully run `npm run build:release` locally? Yes

2. How did you test these changes? Added Unit and Visual tests and manually verified

3. If you made changes to the component library, have you provided corresponding documentation changes? yes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/components/ui/Chat/MessageAttachment/Styled.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import styled from 'styled-components';
+import { baseStyles, baseSpacing } from '../../Base';
+
+export const StyledMessageAttachmentContent = styled.div`
+  color: ${(props) => props.theme.colors.greys.grey80};
```
Comment (bradmanAWS) [inline src/components/ui/Chat/MessageAttachment/Styled.tsx:8]: We will want to put color properties into the light and dark theme files, even if we don't know the dark theme value.

File: src/components/ui/Chat/MessageAttachment/Styled.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  display: flex;
+  flex-direction: row;
+  width: fit-content;
+  padding: 10px;
+  border-radius: 4px;
+  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.1);
+  background-color: ${(props) => props.theme.messageAttachment.content.gbd};
```
Comment (bradmanAWS) [inline src/components/ui/Chat/MessageAttachment/Styled.tsx:15]: sp: bgd

File: src/theme/styled.d.ts [hunk 3]
```diff
@@ -284,6 +284,27 @@ declare module 'styled-components' {
...
       }
     }
+
+    messageAttachment: {
+      size: {
+        fontColor: string;
+        bgd: string;
+        letterSpacing: string;
+        lineHight: string;
+        fontSize: string;
```
Comment (bradmanAWS) [inline src/theme/styled.d.ts:294]: If these values match the BaseStyles values we don't need them here.

File: src/components/ui/Chat/MessageAttachment/Styled.tsx (hunk 1 above)
Comment (nicklera1) [reply src/components/ui/Chat/MessageAttachment/Styled.tsx:8]: oops missed it! Thanks

File: src/components/ui/Chat/MessageAttachment/Styled.tsx (hunk 2 above)
Comment (nicklera1) [reply src/components/ui/Chat/MessageAttachment/Styled.tsx:15]: thanks!

File: src/theme/styled.d.ts (hunk 3 above)
Comment (nicklera1) [reply src/theme/styled.d.ts:294]: Yeah! This do not! it's very specific.

**Review by bradmanAWS — APPROVED** [review]

**Pull request merged**